From c2bd5f28cf634fb9f63ab2db698035a70cc13680 Mon Sep 17 00:00:00 2001
From: John Brooks <john.brooks@jollamobile.com>
Date: Tue, 4 Dec 2012 19:01:57 -0700
Subject: [PATCH] Always allow accounts with always_dispatch to connect

The always_dispatch account setting historically would allow an account
to connect without regard to connectivity, which is needed by the ring
manager. Kludge-transport did not obey that setting, and would interfere
with connections on those accounts.

This allows the ring manager to coexist with a connectivity backend, and
maintain its connection even when not connected to the internet.
---
 src/kludge-transport.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/kludge-transport.c b/src/kludge-transport.c
index c8e2ba1..66b98e3 100644
--- a/src/kludge-transport.c
+++ b/src/kludge-transport.c
@@ -235,7 +235,13 @@ mcd_kludge_transport_account_connection_cb (
   McdKludgeTransport *self = MCD_KLUDGE_TRANSPORT (user_data);
   McdKludgeTransportPrivate *priv = self->priv;
 
-  if (mcd_connectivity_monitor_is_online (priv->minotaur))
+  if (_mcd_account_needs_dispatch (account))
+    {
+      DEBUG ("Always-dispatchable account %s needs no transport, proceeding",
+          mcd_account_get_unique_name (account));
+      mcd_account_connection_proceed (account, TRUE);
+    }
+  else if (mcd_connectivity_monitor_is_online (priv->minotaur))
     {
       mcd_account_connection_bind_transport (account, (McdTransport *) self);
       mcd_account_connection_proceed (account, TRUE);
-- 
1.7.9.5

